docs/fsharp/whats-new/fsharp-9.md

```diff
@@ -0,0 +1,432 @@
+
+There are four functions (each with three variants) available:
+
+#### Shuffle
+
+The shuffle functions return a new collection of the same type and size, with each item in a randomly mixed position. The chance to end up in any position is weighted evenly on the length of the collection.
+
+```fsharp
+let allPlayers = [ "Alice"; "Bob"; "Charlie"; "Dave" ]
+let round1Order = allPlayers |> List.randomShuffle // [ "Charlie"; "Dave"; "Alice"; "Bob" ]
+```
+
+For arrays, there are also `InPlace` variants that shuffle the array in place.
+
+#### Choice
+
+The choice functions return a single random element from the given collection. The random choice is weighted evenly on the size of the collection.
+
+```fsharp
+let allPlayers = [ "Alice"; "Bob"; "Charlie"; "Dave" ]
+let randomPlayer = allPlayers |> List.randomChoice // "Charlie"
+```
+
+#### Choices
+
+Choices select N elements from the input collection in random order, allowing elements to be selected more than once.
+
+```fsharp
+let weather = [ "Raining"; "Sunny"; "Snowing"; "Windy" ]
+let forecastForNext3Days = weather |> List.randomChoices 3 // [ "Windy"; "Snowing"; "Windy" ]
+```
+
+#### Sample
+
+Sample selects N elements from the input collection in random order, without allowing elements to be selected more than once. N cannot be greater than the collection length.
+
+```fsharp
+let foods = [ "Apple"; "Banana"; "Carrot"; "Donut"; "Egg" ]
+let today'sMenu = foods |> List.randomSample 3 // [ "Donut"; "Apple"; "Egg" ]
+```
+
+For a full list of functions and their variants, see ([RFC #1135](https://github.com/fsharp/fslang-design/blob/main/RFCs/FS-1135-random-functions-for-collections.md)).
+
+### Parameterless constructor for `CustomOperationAttribute`
+
+This makes it easier to create a custom operation for a computation expression builder. It will use the name of the method instead of having to explicitly name it (when in most cases the name matches the method name already).
+
+```fsharp
+type FooBuilder() =
+    [&lt;CustomOperation&gt;]  // Previously would have to be [&lt;CustomOperation("bar")&gt;]
+    member _.bar(state) = state
+```
+
+### C# collection expression support for F# lists and sets
+
+When using F# lists and sets from C#, you can now use collection expressions to initialize them.
+
+Instead of:
+
+<pre><code class="language-csharp">FSharpSet&lt;int&gt; mySet = SetModule.FromArray([1, 2, 3]);
+```
+
+You can now write:
+
+<pre><code class="language-csharp">FSharpSet&lt;int&gt; mySet = [ 1, 2, 3 ];
+```
+
+This makes it easier to use the F# immutable collections from C#, for example when you need their structural equality, which `System.Collections.Immutable` collections don't have.
+
+## Quality of life improvements
+
+### Parser recovery
+
+There have been continuous improvements in parser recovery, meaning that tooling (e.g. syntax highlighting) still works with code when you're in the middle of editing it and it may not be syntactically correct at all times.
```

Comment:
```suggestion
There have been continuous improvements in parser recovery, meaning that tooling (for example, syntax highlighting) still works with code when you're in the middle of editing it and it might not be syntactically correct at all times.
```